common/static/common/js/discussion/views/discussion_inline_view.js

```diff
@@ -197,20 +198,18 @@
                 return;
             }
             if (this.showed) {
-                this.newPostForm.slideDown(300);
-            } else {
-                this.newPostForm.show().focus();
+                this.$('section.discussion').find('.inline-discussion-thread-container').hide();
+                this.newPostForm.removeClass('is-hidden').focus();
             }
             this.newPostView.$el.removeClass('is-hidden');
             this.toggleDiscussionBtn.addClass('shown');
-            this.toggleDiscussionBtn.find('.button-text').html(gettext('Hide Discussion'));
-            this.$('section.discussion').slideDown();
+            edx.HtmlUtils.setHtml(this.toggleDiscussionBtn.find('.button-text'), gettext('Hide Discussion'));
             this.showed = true;
         },
 
         hideNewPost: function() {
-            this.newPostView.$el.addClass('is-hidden');
-            return this.newPostForm.slideUp(300);
+            this.$('section.discussion').find('.inline-discussion-thread-container').show();
```

Comment:
We shouldn't mix using ``.show`` and the ``is-hidden`` class. My preference is to use ``is-hidden`` for all situations where we want to hide/show something.